### ISSUE TITLE
Losses: flow matching target without a model

### ISSUE THREAD

**Issue by soran-ghaderi** [soran-ghaderi/torchebm#331]
Sub-task of ##328.
Today:

    loss = FlowMatchingLoss(model, interpolant=...)
    value = loss(x)

Wanted (names are a sketch):

    batch = prepare_flow_matching(x1, x0=None, t=None, generator=g)  # xt, t, target
    pred = my_model(batch.xt, batch.t)
    value = flow_matching_loss(pred, batch)

The logic lives in FlowMatchingLoss.compute_loss / training_losses and EquilibriumMatchingLoss.training_losses. Move it out, then have both call it.

Done when: the helper accepts explicit x0, t and generator, and FlowMatchingLoss(model)(x) returns a bit-identical value with the same number of generator draws as before.

**Comment by soran-ghaderi** [comment]
Superseded by the focused subtasks under #334.
